### PR TITLE
Add API endpoint for fetching all metadata for a repository.

### PR DESCRIPTION
**Pull request by davebx**
- ~~Table tracks repository dependencies and is populated when metadata is generated.~~
- API endpoint returns a json dictionary of the requested metadata revision, recursively retrieving repository and tool dependencies it references.

### CODE REVIEW

**Comment by martenson** [comment]
this is waiting for the docker test image update so it can upgrade the db

**Comment by davebx** [comment]
This pull request is also relevant to #1392

File: lib/tool_shed/util/shed_util_common.py [hunk 1]
```diff
@@ -365,6 +365,21 @@ def get_tool_dependency_definition_metadata_from_tool_shed( app, tool_shed_url,
...
+    """
+    changeset_tups = []
+    for repository_metadata in repository.downloadable_revisions:
+        ctx = hg_util.get_changectx_for_changeset( repo, repository_metadata.changeset_revision )
+        if ctx:
+            rev = ctx.rev()
+        else:
+            rev = -1
+        changeset_tups.append( ( rev, repository_metadata.changeset_revision ) )
+    return sorted( changeset_tups )
```
Comment (nsoranzo) [inline lib/tool_shed/util/shed_util_common.py]: `get_metadata_revisions()` method in this same file seems to already do the same (with more options).

File: lib/tool_shed/util/metadata_util.py [hunk 2]
```diff
@@ -32,6 +71,11 @@ def get_latest_repository_metadata( app, decoded_repository_id, downloadable=Fal
                                                               changeset_revision )
 
 
+def get_metadata_by_id( app, metadata_id ):
```
Comment (nsoranzo) [inline lib/tool_shed/util/metadata_util.py]: This function seems to be unused.

**Comment by nsoranzo** [comment]
Toolshed and/or api tests for this would be nice :blush:

**Comment by davebx** [comment]
@nsoranzo Now there is a test.

**Comment by martenson** [comment]
thanks @davebx !

**Pull request merged**